### DISCUSSION TITLE
Set WETH address of Arbitrum

### DISCUSSION THREAD

**Discussion by mizdler** [wormhole-foundation/wormhole#2896] [Problems / bugs] [answered]
WETH address is unset and causes `wrapAndTransferETH` `completeTransferAndUnwrapETH` functions to fail.

I believe `[0x82af49447d8a07e3bd95bd0d56f35241523fbab1](https://arbiscan.io/address/0x82af49447d8a07e3bd95bd0d56f35241523fbab1)` is the canonical wrapped ETH contract on Aribtrum.

I suggest setting it as the WETH address.

**Comment by evan-gray** [comment]
@mizdler These functions purposefully fail on Arbitrum. That is because `ETH` on Arbitrum is already wrapped by the Arbitrum bridge. Sending it via Wormhole would result in double-wrapped (Arbitrum-bridged, Portal-wrapped) `ETH` and would not be fungible with WETH anywhere else.

**Comment by mizdler** [comment]
@evan-gray Thank for your quick reply.

Let's say we a have pool of this double-wrapped ETH with ETH on Solana
Users can swap and receive the native currency of Arbitrum. which I think is a valid use case.

Why not leaving this decision to integrators? 
If Portal doesn't want users to swap token X I think it's better to prevent it on dApp level instead of contract level.

**Comment by evan-gray** [comment]
Thanks for the rationale. I’ll reopen this and circulate with the contributors to see if we can get consensus and make a recommendation.

**Comment by tonyjin** [comment]
Hey @mizdler, @evan-gray brought this up and we plan to have a larger discussion with contributors. We'll have to delay that chat for a couple of weeks due to urgent high priority work that many of us are focusing on this and next week.

**Comment by evan-gray** [comment]
Hey @mizdler, just wanted to let you know that an upgrade to set this is planned to go out for governance today 🙂

**Comment by evan-gray** [comment, marked as answer]
@mizdler governance passed and the WETH address is now set! 🙌
